src/components/PeopleTable.tsx

```diff
@@ -1,683 +1,233 @@
-        </tr>
-
-        <tr data-cy="person">
-          <td>
-            <a
-              className="has-text-danger"
-              href="#/people/maria-sturm-1835"
-            >
-              Maria Sturm
-            </a>
-          </td>
-          <td>f</td>
-          <td>1835</td>
-          <td>1917</td>
-          <td>Seraphina Spelier</td>
-          <td>Charles Sturm</td>
-        </tr>
+import classNames from 'classnames';
+import { FC } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  listOfPeople: Person[];
+};
 
-        <tr data-cy="person">
-          <td>
-            <a
-              className="has-text-danger"
-              href="#/people/emma-de-milliano-1876"
-            >
-              Emma de Milliano
-            </a>
-          </td>
-          <td>f</td>
-          <td>1876</td>
-          <td>1956</td>
-          <td>Sophia van Damme</td>
-          <td>Petrus de Milliano</td>
-        </tr>
+export const PeopleTable: FC<Props> = ({ listOfPeople }) => {
+  const { selectedSlug = '' } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+
+  const list = listOfPeople.map(person => {
+    const mother = listOfPeople.find(m => m.name === person.motherName);
+    const father = listOfPeople.find(f => f.name === person.fatherName);
+
+    return {
+      ...person,
+      mother,
+      father,
+    };
+  });
+
+  let copyList = [...list];
+
+  const sortOptions = (sortlist: Person[]) => {
+    sortlist.sort((a, b) => {
+      let personA = a;
+      let personB = b;
+
+      if (order === 'desc') {
+        [personA, personB] = [personB, personA];
+      }
+
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return personA[sort].localeCompare(personB[sort]);
+        case 'born':
+        case 'died':
+          return personA[sort] - personB[sort];
+        default:
+          return 0;
+      }
+    });
+  };
+
+  const filterListByPeopleFilters = () => {
+    if (centuries.length) {
+      copyList = copyList.filter(person => (
+        centuries.includes((Math.floor(person.born / 100) + 1).toString())
+      ));
+    }
+
+    if (sex) {
+      copyList = copyList.filter(person => (person.sex === sex));
+    }
+
+    if (query) {
+      copyList = copyList.filter(person => (
+        person.name.toLowerCase().includes(query.toLowerCase())
+        || person.motherName?.toLowerCase().includes(query.toLowerCase())
+        || person.fatherName?.toLowerCase().includes(query.toLowerCase())
+      ));
+    }
+  };
+
+  filterListByPeopleFilters();
+
+  sortOptions(copyList);
```

Comment:
Could you use hooks for it? It can lead to a memory leak. Call these functions only when you need to (useEffect). And wrap them into useCallback() if you will use them in the dependency array